pkg/controllers/certificate/cert_rotation_controller_test.go

```diff
@@ -0,0 +1,455 @@
+    server: https://192.168.0.180:56016
+  name: kind-member1
+contexts:
+- context:
+    cluster: kind-member1
+    user: kind-member1
+  name: member1
+current-context: member1
+kind: Config
+preferences: {}
+users:
+- name: kind-member1
+  user:
+    client-certificate-data: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURJVENDQWdtZ0F3SUJBZ0lJWWR6SWRFMFphbEF3RFFZSktvWklodmNOQVFFTEJRQXdGVEVUTUJFR0ExVUUKQXhNS2EzVmlaWEp1WlhSbGN6QWVGdzB5TkRBNE1UVXdOakUyTURCYUZ3MHlOVEE0TVRVd05qRTJNREZhTURReApGekFWQmdOVkJBb1REbk41YzNSbGJUcHRZWE4wWlhKek1Sa3dGd1lEVlFRREV4QnJkV0psY201bGRHVnpMV0ZrCmJXbHVNSUlCSWpBTkJna3Foa2lHOXcwQkFRRUZBQU9DQVE4QU1JSUJDZ0tDQVFFQXFWSGZiUUVxdHVhdHYxU3IKeEtXaTZrVTJpa0FDdTA4eFJJK2JCU1ZyNjZkbEpZYklVb2ozR0hoRTlXV3h0bVlTRjVjNU90eGJraVRiWVcyeAoxRjZ5RHAzS21SRE9VdXJreGFBSDFjNW5wbFFrYy9mM1lFVGpuTlpKUGtvTjhnWkpwclpud2FaU0lWNXppWW9ECm5rKzZ6NjJEM25pQ1ZCNjlCaDdVN0xNMXZlOGRIQXZPc0lJVUNKZ3hQVFVobUZVMEswTEtHQWZJamJHRS85RU4Kd1VhVkpVNGYvQi9aL0ppS20yaU1WMmJWdTVTUG5ubThNYjZXSGNuclVOY3Bpdkw4U0pCdG1aS0V3Z3ZGWkhJZgpMSVJBZmNmYmR5b0w0d2R5dFd3ZHNhcEpXd3A3NG5yRURudGFLSVBjYkFvVmNUSldacmtnaWRnVHFqcTN0WS85CkZwMmNtUUlEQVFBQm8xWXdWREFPQmdOVkhROEJBZjhFQkFNQ0JhQXdFd1lEVlIwbEJBd3dDZ1lJS3dZQkJRVUgKQXdJd0RBWURWUjBUQVFIL0JBSXdBREFmQmdOVkhTTUVHREFXZ0JReUJVbXNXT08rV1ZXVFdjTFlLZlRLOFQ1RQpuekFOQmdrcWhraUc5dzBCQVFzRkFBT0NBUUVBSGQ5YU92Q0pRcFBYdzdQSmE2bSs3ZUNYSmxUdy9lZXFhT1ptCkUvU3F5MER5U0hLalptZWR4WjFHdTlEL2Q0Qnl6Ri82VUFGTjArSjZ5bUdZa2Y3TzNqempCSmh6a2U1UzRzbHMKWUpZU290cWd1VW5YakpCTVJNZzl0YUpSY1RiSU9oVmNHdDByd2t2NDB6V3JpdEplTTFEYmV0Q3FORm9xcms0bQo1WWszWHVwT01VejRtb0dnUzVDbmdIS0xXckVMRjI5VWFsM1BXRUdRbnZoVEc0N2dtUmk2dVVTUGNreGJnVm4wCithVU5Rei9Yc0tpOXgwUEtQU2xJTzMrMFVaMTdIa3Zlb1ZHOGdXRkR5bU94ckMzcEZHUkNNWTFJM3RVdUtvMTYKVUI5NW1XSFFUNjlBZllyanh3R3J4QzdkNTRKL1Y1NERnSUF6cVhFT2FFK1pKTHVYTFE9PQotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
```

Comment:
The certificate will expire on August 15 next year, which may cause this use case to fail. Can you provide a certificate with a longer validity period, and add comments to indicate the expiration time?

Reply:
Is this test case necessary, considering it is effectively covered by `func Test_getKubeconfigFromSecret` and `func TestCertRotationController_shouldRotateCert`?

Reply:
oh yes, thinks for you suggestion~    let me handle it